### ISSUE TITLE
function-call-argument-newline work when function goes across multiple lines

### ISSUE THREAD

**Issue by mitchellwarr** [eslint/eslint#12794]
**What rule do you want to change?**
function-call-argument-newline

**Does this change cause the rule to produce more or fewer warnings?**
more

**How will the change be implemented? (New option, new default behavior, etc.)?**
new option

**Please provide some example code that this change will affect:**

```js

foo("one", "two", "three");
// or
foo(
   "one", 
   "two",
   "three"
);
// not
foo("one", "two",
  "three");

bar("one", "two", { one: 1, two: 2 });
// or
bar(
  "one",
  "two",
  {
     one: 1,
     two: 2
  }
);
// not
bar("one", "two", {
    one: 1,
    two: 2
});

baz(
  "one",
  "two",
   (x) => {
      console.log(x);
   }
);
// not
baz("one", "two", (x) => {
    console.log(x);
});
```

**What does the rule currently do for this code?**
Currently this rule will require a newline for arguments "always", "never", or "consistent" which means either all args are on a newline or all are on the same line.

**What will the rule do after it's changed?**
With a multi line rule, it should look if the parentheses are on different lines. Currently this is valid:
``` js
baz("one", "two", (x) => {
    console.log(x);
});
``` 
because it sees it as all being on one line (i suspect) but there should be a way to say that "no, these should each have their own line"

**Are you willing to submit a pull request to implement this change?**
I dont really understand how eslint works and I dont have a lot of time. Maybe if I get time.

**Comment by kaicataldo** [comment]
It sounds like you're looking for an option like `array-bracket-newline`'s [`multiline` option](https://eslint.org/docs/rules/array-bracket-newline#options). From the docs (emphasis my own):

> `"multiline": true` (default) requires line breaks **if there are line breaks inside elements** or between elements. If this is false, this condition is disabled.

This seems like a reasonable addition to me! I've put my 👍 of support.

**Comment by mdjermanovic** [comment]
Hi @mitchellwarr, thanks for the issue!

I think this is a generally reasonable proposal, but there are some details that probably need discussion.

In particular, this is different from how "multiline" options work in some similar rules, like in the [array-element-newline](https://eslint.org/docs/rules/array-element-newline) rule (I believe that's the most similar rule). There, the condition doesn't depend on `[]` brackets, but on the multiline elements (linebreaks inside elements, not between elements). Also, this proposal allows multiple valid choices for the same code.

It would be nice (although not necessary) to have equivalent options in similar rules. On the other hand, this proposal as it is now might make more sense than the existing options in other rules.

**Comment by kaicataldo** [comment]
Sorry, I should have been clearer. I'm generally in support of adding an option like this, but do agree with @mdjermanovic that we should try to align the API with existing rules for consistency.
